module_fcst_grid_comp.F90

```diff
@@ -737,16 +703,8 @@ subroutine fcst_initialize(fcst_comp, importState, exportState, clock, rc)
       call get_atmos_model_ungridded_dim(nlev=numLevels,         &
                                          nsoillev=numSoilLayers, &
                                          ntracers=numTracers)
-!
-!-----------------------------------------------------------------------
-!
-      IF(rc /= ESMF_SUCCESS) THEN
```

Comment:
I think it's good to report if the fcst_initialize fails.

Reply:
This is not used consistently, for example in other routines like fcst_run_phase_1 and fcst_run_phase_1 this was commented out. Moreover if this routine fails, i.e. rc is not ESMF_SUCCESS at this moment calling code will report an error. In addition as soon as one of the preceding ESMF calls return we check rc and handle that error and return early, so this final check is probably never reached in case of an error.